students/DennisLee/lesson07/test_html_render.py

```diff
@@ -0,0 +1,691 @@
+            self.assertEqual(self.strs_out[3], f'  <h3>{consts.strs_after[2]}</h3>\n')
+            self.assertEqual(self.strs_out[4], f'  <h4>{consts.strs_after[3]}</h4>\n')
+            self.assertEqual(self.strs_out[5], f'  <h5>{consts.strs_after[4]}</h5>\n')
+            self.assertEqual(self.strs_out[6], f'  <p>\n')
+            self.assertEqual(self.strs_out[7], f'    Text under heading 5.\n')
+            self.assertEqual(self.strs_out[8], f'  </p>\n')
+            self.assertEqual(self.strs_out[9], 
+                    f'  <h6 alt="Some hover text">{consts.strs_after[5]}</h6>\n')
+            self.assertEqual(self.strs_out[10],  '</body>\n')
+        
+    # Test `meta` self-closing element code
+    def test_Meta_100(self):  # Can't initialize w/int content
+        with self.assertRaises(TypeError):
+            self.e1 = hr.Meta(50)
+    def test_Meta_101(self):  # Can't initialize w/float content
+        with self.assertRaises(TypeError):
+            self.e1 = hr.Meta(70.5)
+    def test_Meta_110(self):  # Can't initialize w/string content
+        with self.assertRaises(TypeError):
+            self.e1 = hr.Meta("UTF-8")
+    def test_Meta_120(self):  # Can't append a string
+        self.e1 = hr.Meta()
+        with self.assertRaises(TypeError):
+            self.e1.append("   Some \t text.   \n")
+    def test_Meta_130(self):  # Can't append an element
+        self.e2 = hr.P("\n    Nothing. \t\t")
+        self.e1 = hr.Meta()
+        with self.assertRaises(TypeError):
+            self.e1.append(self.e2)
+    # Verify `meta` rendering + attribute within its parent
+    def test_Meta_200(self):  
+        self.e2 = hr.Meta(charset="\t    UTF-8   \n")
+        self.e1 = hr.Head(self.e2)
+        with open(consts.test_filename, 'w') as self.fobj:
+            self.assertTrue(self.e1.render(self.fobj, '     '))
+        with open(consts.test_filename, 'r') as self.file:
+            self.strs_out = self.file.readlines()
+            self.assertEqual(len(self.strs_out), 3)  # Verify total lines
+            self.assertEqual(self.strs_out[0], '<head>\n')
+            self.assertEqual(self.strs_out[1], '     <meta charset="UTF-8" />\n')
+            self.assertEqual(self.strs_out[2], '</head>\n')
+
+    # Test a multilayered `html` document stream w/DOCTYPE declared.
+    def test_Html_with_doctype_100(self):
+        import html_render as hr  # This repeated statement shouldn't
+                                  # be needed, but I keep getting errors
+                                  # without it here
+        meta = hr.Meta(charset="\t    UTF-8   \n")
+        title = hr.Title("\n\nRandom  \t   Page   \n\t   ")
+        head = hr.Head([meta, title])
+        li_1 = hr.Li("  First  item:  ")
+        para = hr.P("\n\nExplanatory\t\ttext.\n\n", **consts.attrs_pass)
+        link = hr.A("https://github.com/", "A useful site!")
+        para.append(link)
+        hr = hr.Hr()
```

Comment:
you were getting an import error because you are overwriting your html_render module by assigning instance to `hr`

I would also avoid aliases unless there is a really good reason for it, as it reduces readability and as you can see here can also be more likely to collide with var names.

Reply:
I see, yes I'm not a fan of aliases generally 🙂